src/controllers/cart.js

```diff
@@ -74,41 +75,38 @@ exports.putCartItemByDishId = async (req, res, next) => {
     }
     return res.status(201).json(cart);
   } catch (err) {
-    return next(new CustomError(err.message, 500));
+    return next(new CustomError(err.message));
   }
 };
 
-exports.getCartItemByDishId = async (req, res) => {
+exports.getCartItemByDishId = async (req, res, next) => {
   try {
     const cart = await Cart.findOne({
       customerId: req.user.id,
       'cartItems.dishId': req.params.id,
     });
     if (cart.cartItems[0]) {
-      res.status(200).json(cart.cartItems[0]);
-    } else {
-      res.status(404).json({ message: 'item not found' });
+      return res.status(200).json(cart.cartItems[0]);
     }
+    return res.status(404).json({ message: 'item not found' });
   } catch (err) {
-    res.status(500).json({ message: 'Failed to get item', error: err });
+    return next(new CustomError(err.message));
   }
 };
 
-exports.deleteCartItemByDishId = async (req, res) => {
+exports.deleteCartItemByDishId = async (req, res, next) => {
   try {
     const cart = await Cart.findOne({
       customerId: req.user.id,
       'cartItems.dishId': req.params.id,
     });
-
-    if (cart.cartItems[0]) {
+    if (cart) {
       cart.cartItems.splice(0, 1);
       cart.save();
-    } else {
-      res.status(404).json({ message: 'item not found' });
+      return res.status(204).json(cart);
```

Comment:
why did you remove ```catch (err) {
    return next(new CustomError(err.message, 422));```
if the time was not exists you should an error message

Reply:
it's in line 108, I did code refactoring